src/main/java/com/chnu/util/LoggedUserUtil.java

```diff
@@ -0,0 +1,19 @@
+package com.chnu.util;
+
+import com.chnu.exception.UserNotFoundException;
+import com.sun.security.auth.UserPrincipal;
+import org.springframework.security.core.context.SecurityContextHolder;
+import java.util.Optional;
+
+public class LoggedUserUtil {
+
+    public static Optional<UserPrincipal> getLoggedUser() {
```

Comment:
We don't need to return instance of UserPrincipal class. We need to return instance of User class.